cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs

```diff
@@ -316,22 +319,22 @@ renderScriptIntegrityHash Nothing = Aeson.Null
 renderScriptHash :: ScriptHash StandardCrypto -> Text
 renderScriptHash = Api.serialiseToRawBytesHexText . Api.fromShelleyScriptHash
 
-renderMissingRedeemers :: [(Alonzo.ScriptPurpose StandardCrypto, ScriptHash StandardCrypto)] -> Aeson.Value
-renderMissingRedeemers scripts = Aeson.object $ map renderTuple  scripts
+renderMissingRedeemers :: Typeable era => Proxy era -> [(Alonzo.ScriptPurpose StandardCrypto, ScriptHash StandardCrypto)] -> Aeson.Value
+renderMissingRedeemers proxy scripts = Aeson.object $ map renderTuple  scripts
  where
   renderTuple :: (Alonzo.ScriptPurpose StandardCrypto, ScriptHash StandardCrypto) -> Aeson.Pair
   renderTuple (scriptPurpose, sHash) =
-    Aeson.fromText (renderScriptHash sHash) .= renderScriptPurpose scriptPurpose
+    Aeson.fromText (renderScriptHash sHash) .= renderScriptPurpose proxy scriptPurpose
 
-renderScriptPurpose :: Alonzo.ScriptPurpose StandardCrypto -> Aeson.Value
-renderScriptPurpose (Alonzo.Minting pid) =
+renderScriptPurpose :: forall era. Typeable era => Proxy era -> Alonzo.ScriptPurpose StandardCrypto -> Aeson.Value
+renderScriptPurpose Proxy (Alonzo.Minting pid) =
   Aeson.object [ "minting" .= toJSON pid]
-renderScriptPurpose (Alonzo.Spending txin) =
+renderScriptPurpose Proxy (Alonzo.Spending txin) =
   Aeson.object [ "spending" .= Api.fromShelleyTxIn txin]
-renderScriptPurpose (Alonzo.Rewarding rwdAcct) =
+renderScriptPurpose Proxy (Alonzo.Rewarding rwdAcct) =
   Aeson.object [ "rewarding" .= Aeson.String (Api.serialiseAddress $ Api.fromShelleyStakeAddr rwdAcct)]
-renderScriptPurpose (Alonzo.Certifying cert) =
-  Aeson.object [ "certifying" .= toJSON (Api.textEnvelopeDefaultDescr $ Api.fromShelleyCertificate cert)]
+renderScriptPurpose Proxy (Alonzo.Certifying cert) =
+  Aeson.object [ "certifying" .= toJSON (Api.textEnvelopeDefaultDescr $ Api.fromShelleyCertificate @era cert)]
```

Comment:
I'm unable to access the `era` type here.  I tried taking a proxy for getting the `era` type but there is no `era` type in the caller: `instance ToJSON (Alonzo.CollectError StandardCrypto) where`

I think this is not an issue with new versions of ledger because the type `Collector` changed from `data CollectError c` to `data CollectError era`.